### PR TITLE
Clarify when `Is` behaves differently to `PrefixMatches`

### PR DESCRIPTION
**Pull request by WillSewell**
Based on a quick read of the function definition/docs, I had
assumed that `Is` was used to match errors in a stack created by
calls to `Augment`.

This is not the case since `Augment` duplicates the
code from the wrapped error to the wrapping error.

It is only when used in conjunction with `NewInternalWithCause`
that `Is` is useful.

I'm hoping that this documentation helps clarify this. I'm in two
minds about whether it actually helps, so I'd be interested
to here what you think.

### CODE REVIEW

File: errors.go [hunk 1]
```diff
@@ -308,6 +308,8 @@ func Propagate(err error) error {
 // Is checks whether an error is a given code. Similarly to `errors.Is`,
 // this unwinds the error stack and checks each underlying error for the code.
 // If any match, this returns true.
+// Note that Is only behaves differently to PrefixMatches when errors in the stack have different codes.
+// This is the case when errors are initialized with NewInternalWithCause, but not with Augment.
```
Comment (robinjmurphy) [inline errors.go:312]: I think this is a helpful addition. My only concern is that it might go stale if we were to change the behaviour of other functions outside of `Is` e.g. `NewInternalWithCause` and `Augment`. For example, if we added a `terrors.Transform` to change an error code, this comment wouldn't be complete. Then again, the comment isn't saying "this is _only_ the case with `NewInternalWithCause`" so maybe it's ok?

File: errors.go [hunk 2]
```diff
@@ -308,6 +308,8 @@ func Propagate(err error) error {
...
 // We prefer this over using a method receiver on the terrors Error, as the function
 // signature requires an error to test against, and checking against terrors would
 // requite creating a new terror with the specific code.
```
Comment (robinjmurphy) [inline errors.go:315]: I think it would be good to remove the "experimental" warnings on these functions now (I think there's one on `Propagate` and maybe `Augment` too). Happy for that to be a separate PR, or this one, but we're pretty set on these now.

**Review by robinjmurphy — DISMISSED** [review]

File: errors.go (hunk 1 above)
Comment (WillSewell) [reply errors.go]: Yes that is a risk, but I think it's worth it in this case. I've added "For example," to the comment to make it clearer that this might not be the only case in the future.

File: errors.go (hunk 2 above)
Comment (WillSewell) [reply errors.go:315]: I can do this in a separate PR.

Comment (WillSewell) [reply errors.go:315]: I was just trying to push my branch `remove-experimental-warnings` - looks like this already exists! https://github.com/monzo/terrors/pull/31

**Review by robinjmurphy — APPROVED** [review]
